### PR TITLE
Operational insights - add error property to QueryResults

### PR DESCRIPTION
**Pull request by Ohad-Fein**
# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/main/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

File: sdk/operationalinsights/Microsoft.Azure.OperationalInsights/src/Customized/Models/QueryResponseError.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿namespace Microsoft.Azure.OperationalInsights.Models
+{
+    using System.Collections.Generic;
+
+    public class QueryResponseError
```
Comment (pakrym) [inline sdk/operationalinsights/Microsoft.Azure.OperationalInsights/src/Customized/Models/QueryResponseError.cs:5]: Can you please add doc-comments to public members?

**Review by pakrym — APPROVED** [review]

File: sdk/operationalinsights/Microsoft.Azure.OperationalInsights/src/Customized/Models/QueryResponseError.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        /// <summary>
+        /// Error details.
+        /// </summary>
+        public List<QueryResponseError> Details { get; set; }
+        /// <summary>
+        /// A human readable error message.
+        /// </summary>
+        public string Message { get; set; }
+        /// <summary>
+        /// Inner error details if they exist.
```
Comment (ischrei) [inline sdk/operationalinsights/Microsoft.Azure.OperationalInsights/src/Customized/Models/QueryResponseError.cs:20]: no need "they"

File: sdk/operationalinsights/Microsoft.Azure.OperationalInsights/src/Customized/Models/QueryResponseInnerError.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        /// </summary>
+        public string SeverityName { get; set; }
+        /// <summary>
+        /// A human readable error message.
+        /// </summary>
+        public string Message { get; set; }
+        /// <summary>
+        /// A machine readable error code.
+        /// </summary>
+        public string Code { get; set; }
```
Comment (ischrei) [inline sdk/operationalinsights/Microsoft.Azure.OperationalInsights/src/Customized/Models/QueryResponseInnerError.cs:20]: should it be a string or int? how it is returned by Draft?

File: sdk/operationalinsights/Microsoft.Azure.OperationalInsights/src/Customized/Models/QueryResults.cs [hunk 4]
```diff
@@ -38,5 +38,10 @@ public partial class QueryResults
         /// If requested, contains query statistics. See https://dev.loganalytics.io/documentation/Using-the-API/RequestOptions for more info.
         /// </summary>
         public IDictionary<string, object> Statistics{ get; set; }
+
+        /// <summary>
+        /// Contains information of errors.
```
Comment (ischrei) [inline sdk/operationalinsights/Microsoft.Azure.OperationalInsights/src/Customized/Models/QueryResults.cs:43]: "May contain partial query errors"

File: sdk/operationalinsights/Microsoft.Azure.OperationalInsights/src/Customized/Models/QueryResponseError.cs (hunk 2 above)
Comment (Ohad-Fein) [reply sdk/operationalinsights/Microsoft.Azure.OperationalInsights/src/Customized/Models/QueryResponseError.cs:20]: Copied from [here](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/operationalinsights/data-plane/Microsoft.OperationalInsights/preview/2021-05-19_Preview/OperationalInsights.json#L1291).

File: sdk/operationalinsights/Microsoft.Azure.OperationalInsights/src/Customized/Models/QueryResponseInnerError.cs (hunk 3 above)
Comment (Ohad-Fein) [reply sdk/operationalinsights/Microsoft.Azure.OperationalInsights/src/Customized/Models/QueryResponseInnerError.cs:20]: _Severity_ and _SeverityName_ cannot be found in the Swagger file (05-19_Preview) so I chose Severity to be an integer and SeverityName to be a string.
_Code_ is a string by the [05-19_Preview Swagger file](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/operationalinsights/data-plane/Microsoft.OperationalInsights/preview/2021-05-19_Preview/OperationalInsights.json#L1277).

File: sdk/operationalinsights/Microsoft.Azure.OperationalInsights/tests/ScenarioTests/QueryTests.cs [hunk 5]
```diff
@@ -97,6 +97,26 @@ public async Task GetsExceptionWithShortWait()
...
+
+            Assert.Equal(System.Net.HttpStatusCode.OK, response.Response.StatusCode);
+            Assert.Equal("OK", response.Response.ReasonPhrase);
+            Assert.True(response.Body.Tables.Count > 0, "Table count isn't greater than 0");
+            Assert.False(String.IsNullOrWhiteSpace(response.Body.Tables[0].Name), "Table name was null/empty");
+            Assert.True(response.Body.Tables[0].Columns.Count > 0, "Column count isn't greater than 0");
+            Assert.True(response.Body.Tables[0].Rows.Count > 0, "Row count isn't greater than 0");
+
+            Assert.NotNull(response.Body.Error);
+            Assert.Equal("PartialError", response.Body.Error.Code);
```
[the remark this answers is not in the document]
Comment (Ohad-Fein) [reply sdk/operationalinsights/Microsoft.Azure.OperationalInsights/tests/ScenarioTests/QueryTests.cs:117]: The verification can be found on lines 118-120.

**Pull request merged**